src/field/overlay_2_gear_phone.c

```diff
@@ -0,0 +1,513 @@
+#include "unk_02092BE8.h"
+
+typedef struct UnkStruct_02253C86 {
+    u8 unk0;
+    u8 unk1;
+    u16 unk2;
+    u8 unk4;
+    u8 unk5;
+} UnkStruct_02253C86;
+
+static const UnkStruct_02253C86 ov02_02253C84[] = {
+    { 1,  0, 13,  0, 0 },
+    { 1,  0, 7,   0, 0 },
+    { 15, 0, 85,  1, 0 },
+    { 9,  0, 93,  1, 0 },
+    { 2,  0, 0,   0, 0 },
+    { 6,  0, 0,   0, 0 },
+    { 24, 0, 0,   0, 0 },
+    { 24, 0, 142, 1, 0 },
+    { 24, 0, 143, 1, 0 },
+    { 24, 0, 144, 1, 0 },
+    { 24, 0, 145, 1, 0 },
+    { 24, 0, 146, 1, 0 },
+    { 0,  0, 27,  0, 0 }
+};
+
+static u32 ov02_02251FDC(GearPhoneRingManager *gearPhone, PhoneBook *phoneBook, u32 mapId);
+static u32 ov02_022521C0(GearPhoneRingManager *gearPhone, PhoneBook *phoneBook, u32 mapId);
+static u8 ov02_02252218(GearPhoneRingManager *gearPhone, PhoneBook *phoneBook, u32 mapId);
+static void ov02_022522AC(GearPhoneRingManager *gearPhone, BOOL a1);
+
+String *GetPhoneBookEntryName(GearPhoneRingManager *gearPhone, HeapID heapId) {
+    String *str;
+    if (!gearPhone->unk_var0_0 || gearPhone->unk_var2 >= 75) {
+        str = String_New(8, heapId);
+    } else {
+        int phoneMsg     = GetPhoneMessageGmm(gearPhone->unk_var2);
+        MsgData *msgData = NewMsgDataFromNarc(MSGDATA_LOAD_LAZY, NARC_msgdata_msg, phoneMsg, heapId);
+        str              = NewString_ReadMsgData(msgData, 0);
+        DestroyMsgData(msgData);
+    }
+    return str;
+}
+
+void ov02_02251EB8(GearPhoneRingManager *gearPhone, u8 a1, u8 a2, u8 a3, u8 a4, u8 a5) {
+    if (a1 >= 75) {
+        gearPhone->unk_var2    = 0xFF;
+        gearPhone->unk_arr5[0] = 0xFF;
+        gearPhone->unk_arr5[1] = 0;
+        GF_ASSERT(FALSE);
+    } else {
+        gearPhone->unk_var2    = a1;
+        gearPhone->unk_arr5[0] = a2;
+        gearPhone->unk_arr5[1] = a3;
+    }
+
+    gearPhone->unk_var3 = a4;
+    gearPhone->unk_var4 = a5;
+}
+
+u8 ov02_02251EE8(GearPhoneRingManager *gearPhone, u8 *a1) {
+    MI_CpuFill8(a1, 0, 5);
+    if (gearPhone->unk_var2 >= 75) {
+        GF_ASSERT(FALSE);
+        a1[0] = 0xFF;
+        return 0xFF;
+    }
+    a1[0] = gearPhone->unk_arr5[0];
+    a1[1] = gearPhone->unk_arr5[1];
+    a1[3] = gearPhone->unk_var3;
+    a1[4] = gearPhone->unk_var4;
+    a1[2] = gearPhone->unk_var7;
+    return gearPhone->unk_var2;
+}
+
+BOOL ov02_02251F20(GearPhoneRingManager *gearPhone) {
+    Location *position = LocalFieldData_GetCurrentPosition(Save_LocalFieldData_Get(gearPhone->saveData));
+
+    if (MapHeader_GetField14_1E(position->mapId) == 0) {
+        return FALSE;
+    }
+
+    if (gearPhone->unk_var0_0 || gearPhone->unk_var8 < gearPhone->unk_varC) {
+        return FALSE;
+    }
+
+    PhoneBook *phoneBook = AllocAndReadPhoneBook(HEAP_ID_4);
+    u32 var              = ov02_022521C0(gearPhone, phoneBook, position->mapId);
+    if (var) {
+        FreePhoneBook(phoneBook);
+        if (var == 2) {
+            gearPhone->sys->unkD2_7 = 1;
+            StartMapSceneScript(gearPhone->sys, 0x7FF, 0);
```

Comment:
Note for the future: this 0x7FF is probably a std script constant (we don't have one defined yet for this)